backend/routers/issues.py

```
@@ -238,6 +238,55 @@ def upvote_issue(issue_id: int, db: Session = Depends(get_db)):
+
+    # Fetch issues with pagination
+    issues = db.query(Issue).options(defer(Issue.action_plan)).order_by(Issue.created_at.desc()).offset(offset).limit(limit).all()
+
+    # Convert to Pydantic models for validation and serialization
+    data = []
+    for i in issues:
+        # Handle action_plan JSON string
+        action_plan_val = i.action_plan
+        if isinstance(action_plan_val, str) and action_plan_val:
+            try:
+                action_plan_val = json.loads(action_plan_val)
```

Ruff (0.14.14)

[error] Undefined name `json`

(F821)